exporter/prometheus/prometheus_test.go

```diff
@@ -342,3 +342,84 @@ func TestCumulativenessFromHistograms(t *testing.T) {
 		t.Fatalf("\ngot:\n%s\n\nwant:\n%s\n", got, want)
 	}
 }
+
+func TestConstLabelsIncluded(t *testing.T) {
+	key, label := "service", "spanner"
+	exporter, err := NewExporter(Options{
+		ConstLabels: prometheus.Labels{
+			key: label,
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed to create prometheus exporter: %v", err)
+	}
+	view.RegisterExporter(exporter)
+
+	names := []string{"foo", "bar", "baz"}
+
+	var measures mSlice
+	for _, name := range names {
+		measures.createAndAppend("tests/"+name, name, "")
+	}
+
+	var vc vCreator
+	for _, m := range measures {
+		vc.createAndAppend(m.Name(), m.Description(), nil, m, view.Count())
+	}
+
+	if err := view.Register(vc...); err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	defer view.Unregister(vc...)
+
+	view.SetReportingPeriod(time.Millisecond)
+
+	for _, m := range measures {
+		stats.Record(context.Background(), m.M(1))
+	}
+
+	srv := httptest.NewServer(exporter)
+	defer srv.Close()
+
+	var i int
+	var output string
+	for {
+		time.Sleep(10 * time.Millisecond)
+		if i == 1000 {
+			t.Fatal("no output at /metrics (10s wait)")
+		}
+		i++
+
+		resp, err := http.Get(srv.URL)
```

Comment:
```go
resrec := httptest.NewRecorder()
req, _ := http.NewRequest("GET", "https://example.org", nil)
exporter.ServeHTTP(resrec, req)
res := resrec.Result()
body := ioutil.ReadAll(res.Body)
```
should suffice instead of a server

Reply:
I'll replace the other test where I copied this from too.

Reply:
I'm not sure this _will_ suffice. I think we have to ping it a couple of times for metrics to be reported, which is why the loop was there (note the reporting period is 1ms). Is it okay if I leave this in? It breaks the test as suggested.